### PR TITLE
bpo-31415: Improve caching of the importtime option.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue31415

### CODE REVIEW

**Review by brettcannon — APPROVED** [review]

File: Python/import.c [hunk 1]
```diff
@@ -1667,38 +1667,46 @@ PyImport_ImportModuleLevelObject(PyObject *name, PyObject *globals,
...
         static int import_level;
         static _PyTime_t accumulated;
         _Py_IDENTIFIER(importtime);
 
         _PyTime_t t1 = 0, accumulated_copy = accumulated;
 
-        Py_XDECREF(mod);
-
         /* XOptions is initialized after first some imports.
          * So we can't have negative cache.
```
Comment (ncoghlan) [inline Python/import.c]: Is this comment still true? The check for `Py_IsInitialized()` below means that there now *is* a negative cache for `ximporttime == 0`.

**Review by ncoghlan — COMMENTED** [review]
The actual change looks good to me, but there's a now outdated comment that needs to be adjusted to match the new implementation (we know that the -X options will have been processed by the time `Py_IsInitialized()` returns true.

**Comment by warsaw** [comment]
Please double check, but I think I've resolved the conflicts.  I did not remove the comment that @ncoghlan pointed out, but I agree it should probably be removed.

**Comment by ncoghlan** [comment]
If we don't fix the comment in this PR, I suspect the odds of fixing it later are pretty low :)

**Comment by warsaw** [comment]
Agreed!  I tried to keep my change narrowly focused on resolving the conflict.  @serhiy-storchaka I'm happy to make this last change if you want, or you can do it since it's *your* branch :)

**Review by warsaw — APPROVED** [review]

**Pull request merged**